uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerCompanyPortal.java

```diff
@@ -180,19 +182,40 @@ public void enrollDevice(@NonNull final String username,
         // click Sign In button on CP welcome page
         UiAutomatorUtils.handleButtonClick("com.microsoft.windowsintune.companyportal:id/sign_in_button");
 
-        final PromptHandlerParameters promptHandlerParameters = PromptHandlerParameters.builder()
-                .prompt(PromptParameter.LOGIN)
-                .consentPageExpected(false)
-                .expectingLoginPageAccountPicker(false)
-                .sessionExpected(false)
-                .loginHint(null)
-                .build();
-
-        final AadPromptHandler aadPromptHandler = new AadPromptHandler(promptHandlerParameters);
-
-        Logger.i(TAG, "Handle prompt in AAD login page for enrolling device..");
-        // handle AAD login page
-        aadPromptHandler.handlePrompt(username, password);
+        if (isFederated) {
+            final MicrosoftStsPromptHandlerParameters promptHandlerParameters = MicrosoftStsPromptHandlerParameters.builder()
+                    .prompt(PromptParameter.LOGIN)
+                    .consentPageExpected(false)
+                    .expectingLoginPageAccountPicker(false)
+                    .sessionExpected(false)
+                    .isFederated(true)
+                    .loginHint(null)
+                    .build();
+
+            final MicrosoftStsPromptHandler microsoftStsPromptHandler = new MicrosoftStsPromptHandler(promptHandlerParameters);
+
+            Logger.i(TAG, "Handle prompt in ADFS login page for enrolling device..");
+            // handle AAD login page for username
+            UiAutomatorUtils.handleInput("i0116", username);
+            UiAutomatorUtils.handleButtonClick("idSIButton9");
+
+            // handle ADFS login page for password
+            microsoftStsPromptHandler.getLoginComponentHandler().handlePasswordField(password);
```

Comment:
Why can't we just do 
`microsoftStsPromptHandler.handlePrompt(username, password)`

just like we do in the non-federated case?

Reply:
handlePrompt calls handlePasswordField method as well. which has a different resourceId for pwd field from what we see in non-federated case

Reply:
Okay, in that case what you need to do is that for federated case, instead of using `MicrosoftStsPromptHandler` rather use the `AdfsPromptHandler`

Reply:
Yes Shahzaib, I have sent the promptParameters as federated = true to create an object of AdfsPromptHandler only. 
MicrosoftStsPromptHandler takes this boolean and creates an obj of AADHandler or AdfShandler accordingly

Reply:
then in that case it should be working already. I mean internally the AdfsPromptHandler would perform handleInput with the correct resource Id. Is that not working as expected?

Reply:
i think the reason why it was not working earlier is because we don't have an override method of handlePrompt in ADFS Handler. 
 Let's say I add this method and call handleEmail and handlePasswordField which are already present in the ADFS Handler, it will still fail. That is because we first get the CP prompt, which is AAD prompt. Here we still need to use AADHandler's handleEmail method for email. Once I fill this, it will realize that it is a federated user, so takes us to the next screen where we have the username auto-populated but password still needs to be filled. This is the reason why i used UiAutomatorUtils.handleInput("i0116", username); for username but handlePasswordField method of ADFSHandler seperately.

I can add this in ADFSHandler 
void handlePrompt(username, pwd, isFederated) {
// To handle AAD prompt for email
UiAutomatorUtils.handleInput("i0116", username); -> This one is email on AAD screen not ADFS
// To handle ADFS prompt for pwd
handlePasswordField(password);
}

Reply:
We discussed offline that this should be improved and we should be reusing the Login Component Handler and Prompt Handler code for this rather than hard coding resource ids. It can be improved in a different PR or this one. Totally up to @somalaya